h2/src/main/org/h2/engine/Mode.java

```diff
@@ -87,17 +87,16 @@
     public boolean systemColumns;
 
     /**
-     * For unique indexes, NULL is distinct. That means only one row with NULL
-     * in one of the columns is allowed.
+     * <ul>
+     * <li>If {@code 0}, multiple identical indexed columns with at least one
+     * {@code NULL} value are allowed in unique index.</li>
+     * <li>If {@code 1}, multiple identical indexed columns with all {@code NULL}
+     * values are allowed in unique index.</li>
+     * <li>If {@code 2}, multiple identical indexed columns are not allowed in
+     * unique index.</li>
+     * </ul>
      */
-    public boolean uniqueIndexSingleNull;
-
-    /**
-     * When using unique indexes, multiple rows with NULL in all columns
-     * are allowed, however it is not allowed to have multiple rows with the
-     * same values otherwise.
-     */
-    public boolean uniqueIndexSingleNullExceptAllColumnsAreNull;
+    public byte uniquieIndexNullsHandling;
```

Comment:
please use an enum and rename 
   uniquieIndexNullsHandling -> uniqueIndexNullsHandling